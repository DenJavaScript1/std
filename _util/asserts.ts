// Copyright 2018-2022 the Deno authors. All rights reserved. MIT license.
// This module is browser compatible.

/**
 * These assertions exist to create a separation of concerns between internal and testing assertions.
 * All non-testing code within the codebase must use these assertions.
 */

export class DenoStdInternalError extends Error {
  constructor(message: string) {
    super(message);
    this.name = "DenoStdInternalError";
  }
}

/** Make an assertion, if not `true`, then throw. */
export function assert(expr: unknown, msg = ""): asserts expr {
  if (!expr) {
    throw new DenoStdInternalError(msg);
  }
}

/** Use this to assert unreachable code. */
export function unreachable(): never {
  throw new DenoStdInternalError("unreachable");
}
