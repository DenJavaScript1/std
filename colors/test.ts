// Copyright 2018-2019 the Deno authors. All rights reserved. MIT license.
import { test } from "../testing/mod.ts";
import { assertEquals } from "../testing/asserts.ts";
import * as _ from "./mod.ts";
import "../examples/colors.ts";

test(function singleColor(): void {
  assertEquals(_.red("foo bar"), "[31mfoo bar[39m");
});

test(function doubleColor(): void {
  assertEquals(_.bgBlue(_.red("foo bar")), "[44m[31mfoo bar[39m[49m");
});

test(function replacesCloseCharacters(): void {
  assertEquals(_.red("Hel[39mlo"), "[31mHel[31mlo[39m");
});

test(function enablingColors(): void {
  assertEquals(_.getEnabled(), true);
  _.setEnabled(false);
  assertEquals(_.bgBlue(_.red("foo bar")), "foo bar");
  _.setEnabled(true);
  assertEquals(_.red("foo bar"), "[31mfoo bar[39m");
});

test(function testBold(): void {
  assertEquals(_.bold("foo bar"), "[1mfoo bar[22m");
});

test(function testDim(): void {
  assertEquals(_.dim("foo bar"), "[2mfoo bar[22m");
});

test(function testItalic(): void {
  assertEquals(_.italic("foo bar"), "[3mfoo bar[23m");
});

test(function testUnderline(): void {
  assertEquals(_.underline("foo bar"), "[4mfoo bar[24m");
});

test(function testInverse(): void {
  assertEquals(_.inverse("foo bar"), "[7mfoo bar[27m");
});

test(function testHidden(): void {
  assertEquals(_.hidden("foo bar"), "[8mfoo bar[28m");
});

test(function testStrikethrough(): void {
  assertEquals(_.strikethrough("foo bar"), "[9mfoo bar[29m");
});

test(function testBlack(): void {
  assertEquals(_.black("foo bar"), "[30mfoo bar[39m");
});

test(function testRed(): void {
  assertEquals(_.red("foo bar"), "[31mfoo bar[39m");
});

test(function testGreen(): void {
  assertEquals(_.green("foo bar"), "[32mfoo bar[39m");
});

test(function testYellow(): void {
  assertEquals(_.yellow("foo bar"), "[33mfoo bar[39m");
});

test(function testBlue(): void {
  assertEquals(_.blue("foo bar"), "[34mfoo bar[39m");
});

test(function testMagenta(): void {
  assertEquals(_.magenta("foo bar"), "[35mfoo bar[39m");
});

test(function testCyan(): void {
  assertEquals(_.cyan("foo bar"), "[36mfoo bar[39m");
});

test(function testWhite(): void {
  assertEquals(_.white("foo bar"), "[37mfoo bar[39m");
});

test(function testGray(): void {
  assertEquals(_.gray("foo bar"), "[90mfoo bar[39m");
});

test(function testBgBlack(): void {
  assertEquals(_.bgBlack("foo bar"), "[40mfoo bar[49m");
});

test(function testBgRed(): void {
  assertEquals(_.bgRed("foo bar"), "[41mfoo bar[49m");
});

test(function testBgGreen(): void {
  assertEquals(_.bgGreen("foo bar"), "[42mfoo bar[49m");
});

test(function testBgYellow(): void {
  assertEquals(_.bgYellow("foo bar"), "[43mfoo bar[49m");
});

test(function testBgBlue(): void {
  assertEquals(_.bgBlue("foo bar"), "[44mfoo bar[49m");
});

test(function testBgMagenta(): void {
  assertEquals(_.bgMagenta("foo bar"), "[45mfoo bar[49m");
});

test(function testBgCyan(): void {
  assertEquals(_.bgCyan("foo bar"), "[46mfoo bar[49m");
});

test(function testBgWhite(): void {
  assertEquals(_.bgWhite("foo bar"), "[47mfoo bar[49m");
});
