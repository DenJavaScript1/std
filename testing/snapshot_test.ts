// Copyright 2018-2022 the Deno authors. All rights reserved. MIT license.
import { stripColor } from "../fmt/colors.ts";
import { assertInstanceOf, AssertionError, fail } from "./asserts.ts";
import { assertSnapshot } from "./snapshot.ts";

class TestClass {
  a = 1;
  b = 2;
  init() {
    this.b = 3;
  }
  get getA() {
    return this.a;
  }
  func() {}
}

const map = new Map();
map.set("Hello", "World!");
map.set(() => "Hello", "World!");
map.set(1, 2);

Deno.test("Snapshot Test", async (t) => {
  await assertSnapshot(t, { a: 1, b: 2 });
  await assertSnapshot(t, new TestClass());
  await assertSnapshot(t, map);
  await assertSnapshot(t, new Set([1, 2, 3]));
  await assertSnapshot(t, { fn() {} });
  await assertSnapshot(t, function fn() {});
  await assertSnapshot(t, [1, 2, 3]);
  await assertSnapshot(t, "hello world");
});

Deno.test("Snapshot Test - step", async (t) => {
  await assertSnapshot(t, { a: 1, b: 2 });
  await t.step("Nested", async (t) => {
    await assertSnapshot(t, new TestClass());
    await assertSnapshot(t, map);
    await t.step("Nested Nested", async (t) => {
      await assertSnapshot(t, new Set([1, 2, 3]));
      await assertSnapshot(t, { fn() {} });
      await assertSnapshot(t, function fn() {});
    });
    await assertSnapshot(t, [1, 2, 3]);
  });
  await assertSnapshot(t, "hello world");
});

Deno.test("Snapshot Test - Adverse String \\ ` ${}", async (t) => {
  await assertSnapshot(t, "\\ ` ${}");
});

Deno.test("Snapshot Test - Failed Assertion", async (t) => {
  await t.step("Object", async (t) => {
    try {
      await assertSnapshot(t, [1, 2]);
      fail("Expected snapshot not to match");
    } catch (error) {
      assertInstanceOf(error, AssertionError);
      await assertSnapshot(t, stripColor(error.message).split("\n"));
    }
  });
  await t.step("String", async (t) => {
    try {
      await assertSnapshot(t, "Hello!");
      fail("Expected snapshot not to match");
    } catch (error) {
      assertInstanceOf(error, AssertionError);
      await assertSnapshot(t, stripColor(error.message).split("\n"));
    }
  });
});
