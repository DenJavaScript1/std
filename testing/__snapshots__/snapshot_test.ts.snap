export const snapshot = {};

snapshot[`Snapshot Test 1`] = `
{
  a: 1,
  b: 2,
}
`;

snapshot[`Snapshot Test 2`] = `
TestClass {
  a: 1,
  b: 2,
}
`;

snapshot[`Snapshot Test 3`] = `
Map {
  "Hello" => "World!",
  1 => 2,
  [Function] => "World!",
}
`;

snapshot[`Snapshot Test 4`] = `
Set {
  1,
  2,
  3,
}
`;

snapshot[`Snapshot Test 5`] = `
{
  fn: [Function: fn],
}
`;

snapshot[`Snapshot Test 6`] = `[Function: fn]`;

snapshot[`Snapshot Test 7`] = `
[
  1,
  2,
  3,
]
`;

snapshot[`Snapshot Test 8`] = `"hello world"`;

snapshot[`Snapshot Test - step 1`] = `
{
  a: 1,
  b: 2,
}
`;

snapshot[`Snapshot Test - step > Nested 1`] = `
TestClass {
  a: 1,
  b: 2,
}
`;

snapshot[`Snapshot Test - step > Nested 2`] = `
Map {
  "Hello" => "World!",
  1 => 2,
  [Function] => "World!",
}
`;

snapshot[`Snapshot Test - step > Nested > Nested Nested 1`] = `
Set {
  1,
  2,
  3,
}
`;

snapshot[`Snapshot Test - step > Nested > Nested Nested 2`] = `
{
  fn: [Function: fn],
}
`;

snapshot[`Snapshot Test - step > Nested > Nested Nested 3`] = `[Function: fn]`;

snapshot[`Snapshot Test - step > Nested 3`] = `
[
  1,
  2,
  3,
]
`;

snapshot[`Snapshot Test - step 2`] = `"hello world"`;

snapshot[`Snapshot Test - Adverse String \\ \` \${} 1`] = `"\\\\ \` \${}"`;

snapshot[`Snapshot Test - Failed Assertion > Object 1`] = `
[
  1,
  2,
  3,
]
`;

snapshot[`Snapshot Test - Failed Assertion > Object 2`] = `
[
  "Snapshot does not match:",
  "",
  "",
  "    [Diff] Actual / Expected",
  "",
  "",
  "    [",
  "      1,",
  "      2,",
  "+     3,",
  "    ]",
  "",
]
`;

snapshot[`Snapshot Test - Failed Assertion > String 1`] = `"Hello World!"`;

snapshot[`Snapshot Test - Failed Assertion > String 2`] = `
[
  "Snapshot does not match:",
  "",
  "",
  "    [Diff] Actual / Expected",
  "",
  "",
  '-   "Hello!"',
  '+   "Hello World!"',
  "",
  "",
]
`;
